app/code/Magento/CatalogStorefront/Model/VariantService.php

```diff
@@ -0,0 +1,280 @@
+                    ];
+                    //TODO: Adapt to work without store code
+                    $variantsInElasticFormat['product_variant'][self::EMPTY_STORE_CODE][CatalogRepository::SAVE][] =
+                        $variant;
+                }
+            }
+
+            $this->catalogRepository->saveToStorage($variantsInElasticFormat);
+            $importVariantsResponse = $this->importVariantsResponseFactory->create();
+            $importVariantsResponse->setMessage('Records imported successfully');
+            $importVariantsResponse->setStatus(true);
+
+            return $importVariantsResponse;
+        } catch (\Exception $e) {
+            $message = 'Cannot process product variants import: ' . $e->getMessage();
+            $this->logger->error($message, ['exception' => $e]);
+            $importCategoriesResponse = $this->importVariantsResponseFactory->create();
+            $importCategoriesResponse->setMessage($message);
+            $importCategoriesResponse->setStatus(false);
+            return $importCategoriesResponse;
+        }
+    }
+
+    /**
+     * Delete product variants from storage.
+     *
+     * @param DeleteVariantsRequestInterface $request
+     * @return DeleteVariantsResponseInterface
+     */
+    public function DeleteProductVariants(DeleteVariantsRequestInterface $request): DeleteVariantsResponseInterface
+    {
+        $deleteFields = \array_map(function ($id) {
+            return ['id' => $id];
+        }, $request->getId());
+
+        $variantsInElasticFormat = [
+            'product_variant' => [
+                //TODO: Adapt to work without store code
+                self::EMPTY_STORE_CODE => [
+                    CatalogRepository::DELETE_BY_QUERY => $deleteFields
+                ]
+            ]
+        ];
+
+        $deleteVariantsResponse = $this->deleteVariantsResponseFactory->create();
+        try {
+            $this->catalogRepository->saveToStorage($variantsInElasticFormat);
+            $deleteVariantsResponse->setMessage('Product variants were removed successfully');
+            $deleteVariantsResponse->setStatus(true);
+        } catch (\Throwable $e) {
+            $message = 'Unable to delete product variants';
+            $this->logger->error($message, ['exception' => $e]);
+            $deleteVariantsResponse->setMessage($message);
+            $deleteVariantsResponse->setStatus(false);
+        }
+
+        return $deleteVariantsResponse;
+    }
+
+    /**
+     * Get product variants from storage.
+     *
+     * Only variants whose corresponding products are 'enabled' and stored in storage are returned.
+     *
+     * @param ProductVariantRequestInterface $request
+     * @return ProductVariantResponseInterface
+     * @throws \InvalidArgumentException
+     * @throws RuntimeException
+     * @throws \Throwable
+     */
+    public function GetProductVariants(ProductVariantRequestInterface $request): ProductVariantResponseInterface
```

Comment:
I have created configurable product with 2 options:
Color: red, green, blue
Size: s, m, l

result of this method is incorrect

![image](https://user-images.githubusercontent.com/24605579/98623509-76b4c600-22d1-11eb-80c9-cddbc2a5d151.png)

Each variation should have 2 option values, but as we see on the screenshot only half of them have 2 options, the rest of them have 1 option value for some reason.

For id `configurable/32/36` there's only one option value `32:color/Y29uZmlndXJhYmxlLzkzLzE0` however in storage we have 2 records 

![image](https://user-images.githubusercontent.com/24605579/98623389-2f2e3a00-22d1-11eb-965c-726921794c5b.png)

Reply:
Was a bug with max search size. Changed it to 2000 and added a ticket for pagination.